### DISCUSSION TITLE
UDP, TCP, or both ?

### DISCUSSION THREAD

**Discussion by cyrileek** [freenet/freenet-core#854] [Q&A] [answered]
Does Freenet use only UDP protocol like Hyphanet, or does it use TCP too ?
Thanks for all the good work :)

**Comment by ple1n** [comment]
Why does it matter ? Transports are pluggable and I don't think either one is enough. You almost certainly want some baseline censor resistance, which means QUIC/TLS. 

I don't think it concerns any features on transport except for being a byte stream. So It's irrelevant.

**Comment by iduartgomez** [comment, marked as answer]
Currently uses TCP, in the future would like to use QUICK and potentially support other transports.
